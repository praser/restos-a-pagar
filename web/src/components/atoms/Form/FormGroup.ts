import styled from 'styled-components';

interface IProp {
  width?: string;
}

const FormGroup = styled.div<IProp>`
  margin-bottom: 1rem;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
  display: flex;
  flex-direction: column;
  ${({ width }) => width && `width: ${width};`}
  ${({ width }) => !width && `flex: 1;`}

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }
`;

export default FormGroup;
