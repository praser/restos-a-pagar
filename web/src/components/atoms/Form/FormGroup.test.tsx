import React from 'react';

import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import FormGroup from './FormGroup';

describe('FormGroup atom component', () => {
  it('is expected to render the component correctly', () => {
    const formGroup = renderer.create(<FormGroup />).toJSON();
    expect(formGroup).toMatchSnapshot();
  });

  it('is expected to have style flex:1 if no width is provided', () => {
    const { container } = render(<FormGroup />);
    expect(container.firstChild).toHaveStyle('flex: 1');
  });

  it('is expected to have style width: 80% if it is provided', () => {
    const { container } = render(<FormGroup width="80%" />);
    expect(container.firstChild).toHaveStyle('width: 80%');
  });
});
