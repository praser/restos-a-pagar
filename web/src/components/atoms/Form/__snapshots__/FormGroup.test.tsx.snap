// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup atom component is expected to render the component correctly 1`] = `
<div
  className="sc-bdfBwQ jvAeeR"
/>
`;
